### PR TITLE
fix(ui): Add logic to toggle console theme

### LINKED ISSUE #16319: Dark Mode: CSS Declarations hard to read in Editor
#### Challenge Name
Basic CSS: Style Multiple Elements with a CSS Class and beyoned

#### Issue Description

The CSS declarations in dark mode are hard to read. They are somethings navy blue on black.
This also happens for anchor tags in the HTML section.

#### Browser Information

Browser Name, Version:
Operating System: FireFox 57.0 (64-bit) and Chrome Version 63.0.3239.84 (Official Build) (64-bit)
Mobile, Desktop, or Tablet: Laptop Windows 10 Professional 64-bit

#### Your Code

```js



```
#### Screenshot

image

### PR DESCRIPTION
**Pull request by huyenltnguyen**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #16319

#### Description

I made a mistake in my previous PR [(#16496)](https://github.com/freeCodeCamp/freeCodeCamp/pull/16496) by manually changing the console style in the stylesheet instead of implementing the logic to the Output component, which results a terrible render on the live beta (see screenshot).
Same as the previous PR, this one was **not** fully tested due to the syntax highlighting issue.

untitled

### CODE REVIEW

**Comment by raisedadead** [comment]
Just to confirm the current issue only affects night mode? And this PR fixes this? Also, syntax highlight seems to work for me locally.

**Review by raisedadead — APPROVED** [review]
LGTM. I'll wait for @BerkeleyTrue for a look. Also we might need a a border around the console in the day mode.

But, I guess I am a fan of the night mode right now.

**Comment by huyenltnguyen** [comment]
@raisedadead Thanks for the QA. Yes, the issue only affects night mode. It's because I didn't give the console the theme, let it use `default` as a fallback and get all the `default` style (dark text on white background). The syntax highlighting issue made me think the console only displays text, therefore, I decided not to implement any logic to the component as it'd be a bit of an overkill, and I just went tweak the background and text color for night mode instead.
I'm happy to update the PR for the border if you guys need it.
The syntax highlighting issue is weird. A number of people reported that they have this issue as well, but just a few days ago I did see someone's screenshot of the local version, in which the syntax highlighting works fine. It's a bit annoying but not important, though.

**Comment by raisedadead** [comment]
Sure, if you have a moment and the change is not too much work, then please add it while your at this already.

**Comment by huyenltnguyen** [comment]
Added a border for the console in day mode.

File: common/app/routes/Challenges/Output.jsx [hunk 1]
```diff
@@ -13,19 +16,34 @@ const defaultOptions = {
   readOnly: 'nocursor'
 };
 
+const mapStateToProps = createSelector(
```
Comment (BerkeleyTrue) [inline common/app/routes/Challenges/Output.jsx]: There is no need to add a layer of memoization if there is only one value being selected for. In these situations it is fine just to use the selector (in this case the theme selector) with a plain function.

**Review by BerkeleyTrue — CHANGES_REQUESTED** [review]

**Comment by huyenltnguyen** [comment]
Changed as requested.
I also noticed the 'link to github' button doesn't have a green background as others in night mode, so I made a change to fix it too.

untitled

File: common/app/routes/Settings/Social-Settings.jsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ export default function SocialSettings({
     <Button
       block={ true }
       bsSize='lg'
-      className='btn-link-social btn-github'
+      className='btn-link-social btn-github btn-primary'
```
Comment (BerkeleyTrue) [inline common/app/routes/Settings/Social-Settings.jsx]: This change should be a prop

`bsStyle='primary'`

File: common/app/routes/Settings/Social-Settings.jsx [hunk 3]
```diff
@@ -22,6 +22,7 @@ export default function SocialSettings({
     <Button
       block={ true }
       bsSize='lg'
+      bsStyle='primary'
```
Comment (BerkeleyTrue) [inline common/app/routes/Settings/Social-Settings.jsx]: Looking at this now, I don't understand your reasoning for adding this. None of the other social buttons have a primary style. What is missing is the active state. Maybe you where referring to that? In that case the change would be to add `active={ isGithubCool }`

**Comment by BerkeleyTrue** [comment]
I've tested the PR. The output is noticeably improved. I have one more note that requires fixing for this to be merged in.

**Comment by huyenltnguyen** [comment]
@BerkeleyTrue Sorry for all the mess I've made, including the output.
I'm a bit confused, what does `active` mean? Does it mean user has connected to their github account? I didn't really look into its logic, though. The button is just right below the 'toggle theme' button, so while I was working on the theme issue, I noticed that it didn't have a green background like the others (not the other social buttons, but the other buttons on the same page). I didn't see any other social buttons on the page, so there wasn't any style reference for this. That's why I compared it with the 'toggle theme' button and thought it was missing the `primary` class.
If the intent is to light up the button only when github account is connected, then my change is unnecessary and I'll revert it.
Sorry again.

**Comment by systimotic** [comment]
The other social buttons don't have the `primary` style either, but I agree that this is a bit weird. Every single other button on the page does have the `primary` style, even toggles that are off. That said, it's also nice to have a visual difference between an active and an inactive button.
We should probably remove this change from this PR and open a new issue to discuss this.

**Comment by raisedadead** [comment]
> We should probably remove this change from this PR and open a new issue to discuss this.

Up I too agree on this.

**Comment by huyenltnguyen** [comment]
I should have asked first before wasting our time.
Sorry and thanks for all the help!

**Review by BerkeleyTrue — APPROVED** [review]

**Comment by BerkeleyTrue** [comment]
@huyenltnguyen thanks for the PR!

@raisedadead @systimotic The social buttons are styled differently because they have their own styling, which `primary` would override. What is needed is slightly different styling when dark mode is active. Maybe making the background of the button a different shade of gray?

**Comment by raisedadead** [comment]
@huyenltnguyen nice work, trust me I would just default to the night mode..! lol.

**Comment by raisedadead** [comment]
Lets, take Berkeley's advice though, if you are still up for making a PR for the social buttons. But that said Stuart has some things in pipeline and we should probably defer a little on those changes.

**Comment by systimotic** [comment]
Thanks for fixing this @huyenltnguyen, nice work!

To elaborate on what Stuart has in the pipeline, it's this PR: https://github.com/freeCodeCamp/freeCodeCamp/pull/16412
It's massive and touches the area around these buttons, so I agree that it would be a good idea to wait for that.

**Pull request merged**